src/main/java/com/github/manolo8/darkbot/gui/titlebar/BackpageTask.java

```diff
@@ -0,0 +1,264 @@
+                            .start().waitFor();
+                }
+            } catch (InterruptedException | IOException e) {
+                e.printStackTrace();
+            }
+        } // inform user that instance/sid is not valid?
+        SwingUtilities.invokeLater(() -> button.setEnabled(true));
+    }
+
+    private boolean canRun() {
+        try {
+            Version currentVersion = readVersionFile();
+
+            // have downloaded backpage and version was checked recently
+            return (currentVersion != null && VERSION_CHECK_TIMER.isActive())
+                    || checkVersion(currentVersion);
+        } catch (IOException e) {
+            e.printStackTrace();
+            VERSION_CHECK_TIMER.disarm(); // IOException happen - reset timer
+
+            String message = e instanceof DirectoryClearException ? ((DirectoryClearException) e).getPopupMessage() : e.toString();
+            Popups.of("Backpage exception", message)
+                    .messageType(JOptionPane.ERROR_MESSAGE)
+                    .showAsync();
+        } finally {
+            button.removeProgressBar();
+        }
+
+        return false;
+    }
+
+    private boolean checkVersion(Version current) throws IOException {
+        JProgressBar progressBar = button.createProgressBar();
+
+        ReleaseInfo releaseInfo = ReleaseInfo.get();
+        if (releaseInfo == null) return false;
+
+        Version remoteVersion = releaseInfo.getVersion();
+        ReleaseInfo.Asset asset = releaseInfo.getValidAsset();
+        if (remoteVersion == null || asset == null) return false;
+
+        VERSION_CHECK_TIMER.activate(); // activate the timer here - version will be successfully checked
+        return current != null && current.compareTo(remoteVersion) >= 0
+                || askUserToDownload(progressBar, current, remoteVersion, asset);
+    }
+
+    //return true if is possible to run a backpage
+    private boolean askUserToDownload(JProgressBar progressBar,
+                                      Version current, Version remote, ReleaseInfo.Asset asset) throws IOException {
+        String message = current == null ? I18n.get("gui.backpage_button.download_message")
+                : I18n.get("gui.backpage_button.new_version_message");
+        message += "\n -Ver: " + (current != null ? current + " -> " : "") + remote + ", size: " + (asset.size >> 20) + "MB";
+
+        JButton download = new JButton(I18n.get("gui.backpage_button.download"));
+        Popups.Builder builder = Popups.of("Backpage browser", message)
+                .messageType(JOptionPane.QUESTION_MESSAGE)
+                .options(download, I18n.get("gui.backpage_button.cancel"));
+
+        JOptionPane optionPane = builder.build();
+        download.addActionListener(l -> optionPane.setValue(download));
+
+        if (download == builder.showSync()) {
+            if (!clearDirectory())
+                throw new DirectoryClearException(BACKPAGE_PATH);
```

Comment:
This just seems a bit odd, can't just clear directory throw such exception if it needs to?

Creating an exception that gets thrown in one single case and it handled in a generic block is just a bit odd to me, i get the reason here is so that you can show the user something relevant, but couldn't you just use a plain IOException and use the "message" parameter and show that in the pop-up?